code/game/objects/items/weapons/storage/bluespace.dm

```diff
@@ -0,0 +1,84 @@
+/*
+ *		Bag of Holding
+ *		Miniature black hole
+*/
+
+/obj/item/weapon/storage/backpack/holding
+	name = "Bag of Holding"
+	desc = "A backpack that opens into a localized pocket of Blue Space."
+	origin_tech = "bluespace=4"
+	item_state = "holdingpack"
+	icon_state = "holdingpack"
+	fits_max_w_class = 4
+	max_combined_w_class = 28
+
+/obj/item/weapon/storage/backpack/holding/miniblackhole
+	name = "miniature black hole"
+	desc = "A miniature black hole that opens into a localized pocket of Blue Space."
+	icon_state = "porthole"
+	slot_flags = 0
+	w_class = 2 //fits in pockets!
+
+/obj/item/weapon/storage/backpack/holding/suicide_act(mob/user)
+	user.visible_message("<span class = 'danger'><b>[user] puts \the [src.name] on \his head and stretches the bag around \himself. With a sudden snapping sound, the bag shrinks to its original size, leaving no trace of [user].</b></span>")
+	forceMove(get_turf(user))
+	qdel(user)
+
+/obj/item/weapon/storage/backpack/holding/miniblackhole/suicide_act(mob/user)
+	user.visible_message("<span class = 'danger'><b>[user] puts \the [src.name] on the ground and jumps inside, never to be seen again. What is this, a cartoon?<</b></span>")
+	forceMove(get_turf(user))
+	qdel(user)
+
+/obj/item/weapon/storage/backpack/holding/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(W == src)
+		return // HOLY FUCKING SHIT WHY STORAGE CODE, WHY - pomf
+	if(crit_fail)
+		to_chat(user, "<span class = 'warning'>The Bluespace generator isn't working.</span>")
+		return
+	var/obj/item/weapon/storage/backpack/holding/H = locate(/obj/item/weapon/storage/backpack/holding) in W
+	if(H && !H.crit_fail)
+		singulocreate(H, user)
+	if(istype(W, /obj/item/weapon/storage/backpack/holding) && !W.crit_fail)
+		singulocreate(W, user)
+	. = ..()
+
+//BoH+BoH=Singularity, WAS commented out
+/obj/item/weapon/storage/backpack/holding/proc/singulocreate(var/obj/item/weapon/storage/backpack/holding/H, var/mob/user)
+	investigation_log(I_SINGULO,"has become a singularity. Caused by [user.key]")
+	message_admins("[key_name_admin(user)] detonated [H] and [src], creating a singularity.")
+	log_game("[key_name(user)] detonated [H] and [src], creating a singularity.")
+	to_chat(user, "<span class = 'danger'>The Bluespace interfaces of the two devices catastrophically malfunction!</span>")
+	to_chat(user, "<span class='danger'>FUCK!</span>")
+	qdel(H)
+	H = null
+	new /obj/machinery/singularity (get_turf(src))
+	user.throw_at(get_turf(src), 10, 5)
+	qdel(src)
+	return
+
+//Currently unused since reliability isn't changed by anything
```

Comment:
How would be best to implement 'reliability'

What if every object that's put inside above a certain number (Let's say 7) starts to reduce reliability by about 5%, so you can gamble putting a lot of stuff in, but eventually it's all gonna come crashing down